typing/typedtree.mli

```diff
@@ -262,11 +269,15 @@ and expression_desc =
             Texp_record
               { fields = [| l1, Kept t1; l2 Override P2 |]; representation;
                 extended_expression = Some E0 }
+            alloc_mode is the mode of the record (stack/heap)
         *)
-  | Texp_field of expression * Longident.t loc * Types.label_description
+  | Texp_field of expression * Longident.t loc * Types.label_description * Types.alloc_mode
+    (* alloc_mode is the mode of the result; only used when the field is float *)
```

Comment:
Not quite: it's used only when the field is an element of a float-record. Here's a possible comment:

```
(* alloc_mode is the mode of the result; used when getting a field from a record of all floats to decide where to allocate the returned float *)
```

If a record is all floats, OCaml inlines the floats. This means that when extracting a field from such a record, we need to allocate a block for the returned float.

Reply:
thanks - I didn't know that.